src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Control/AsyncModule.fs

```diff
@@ -314,6 +323,84 @@ type AsyncModule() =
 
         Assert.AreEqual(0, counter)
 
+#if FSHARP_CORE_PORTABLE
+// nothing
+#else
+#if FSHARP_CORE_NETCORE_PORTABLE
+// nothing
+#else
+
+    [<Test>]
+    member this.``Async.Choice takes first result that is <> None``() =        
+        let returnFirstResult (n:PositiveInt) (i:PositiveInt) x =
```

Comment:
To avoid casting to `int` afterwards, one can write `(PositiveInt n) (PositiveInt i)`.